src/handlers/records-read.ts

```diff
@@ -37,15 +37,23 @@ export class RecordsReadHandler implements MethodHandler {
       return messageReplyFromError(e, 401);
     }
 
-    // get existing messages matching `recordId` so we can perform authorization
-    const query = {
-      interface : DwnInterfaceName.Records,
-      recordId  : message.descriptor.recordId
+    // get the latest active records matching the supplied filter
+    // only RecordsWrite messages will be returned due to 'isLatestBaseState' being set to true.
+    const query: Filter = {
+      interface         : DwnInterfaceName.Records,
+      isLatestBaseState : true,
+      ...Records.convertFilter(message.descriptor.filter)
     };
-    const existingMessages = await this.messageStore.query(tenant, query) as TimestampedMessage[];
+    const existingMessages = await this.messageStore.query(tenant, query) as RecordsWriteMessage[];
 
-    const newestExistingMessage = await Message.getNewestMessage(existingMessages);
+    // ensure that the returned query only contains a unique record
+    try {
+      RecordsReadHandler.enforceSingleRecordRule(existingMessages);
+    } catch (e) {
+      return messageReplyFromError(e, 400);
```

Comment:
I've been wondering myself what status code we should send. Why 400? My gut tells me this isn't a "bad" request, but it's definitely not a 401. 404 also seems not quite right. Maybe 418? I'll put my musings aside and trust your judgement that 400 is good here. Just want to have it written somewhere if there's a reason.

Reply:
🤣 yeah I had the same thoughts, although not 418 haha!

I landed on 400 because because it feels like a request error from the user's end. They provided some sort of incorrect filter that caused it.

I was thinking 404 at one point, but it felt not descriptive enough since there is a record there, just not only 1.